### ISSUE TITLE
Issue with Home Assistant restart

### ISSUE THREAD

**Issue by matthewvinton** [sdwilsh/brultech-serial2mqtt#253]
I've updated to the latest brultech-serial2mqtt docker image. 

Now when I restart the Home Assistant service, the sensors do not update until I restart the brultech-serial2mqtt docker containers. One I restart those containers the sensors start to update normally.

There are no corresponding log entries when I restart HA. Example:

2024-01-30 18:00:27,045 - brultech_serial2mqtt.mqtt - INFO - Notifying clients that we are online on brultech-serial2mqtt-21843/status

This is from 24 hours previous.

My config.yaml is very vanilla - no modifications to the mqtt config, just basic channel info, COM port info, and broker info.

**Comment by sdwilsh** [comment]
Yeah, this /used/ to work, and I haven't been able to pin down what changed.  I had some changes I landed to try and figure this out and no luck so far :(

**Comment by sdwilsh** [comment]
I finally had some time to get to the bottom of this.  I wrote a comprehensive integration test for it, and got a fix.

I'll be landing that shortly, so we should finally not have to worry about this anymore.
